From fbbe0e0fcf9662cc61bfb084c37dd461a4fdfacb Mon Sep 17 00:00:00 2001
From: Sean Whitton <spwhitton@spwhitton.name>
Date: Thu, 22 Dec 2016 17:31:05 +0000
Subject: [PATCH] dgit-sponsorship(7): Use --no-dep14tag

While this complicates the sponsor's workflow a little, it simplifies
the sponsee's workflow: they don't have to decide whether to delete
DEP-14 tags that other guides have told them to create.

Signed-off-by: Sean Whitton <spwhitton@spwhitton.name>
---
 dgit-sponsorship.7.pod | 30 +++++++++++++-----------------
 1 file changed, 13 insertions(+), 17 deletions(-)

diff --git a/dgit-sponsorship.7.pod b/dgit-sponsorship.7.pod
index 3fc59d2..8d5b72d 100644
--- a/dgit-sponsorship.7.pod
+++ b/dgit-sponsorship.7.pod
@@ -107,8 +107,6 @@ alioth is another possibility.
 The branch names used by the sponsee on their local machine,
 and on the server, do not matter.
 
-The sponsee should not make a C<debian/>I<version> tag.
-
 Instead, the sponsee should include the
 git commit id of their HEAD
 in their handover email.
@@ -197,12 +195,6 @@ Check the git commit ID of the sponsee's branch tip,
 and the sha256sums of the .origs,
 against the handoff email.
 
-Confirm that the sponsee has not made
-a debian/1.2.3-1 tag.
-If they have,
-it is best to ask them to delete it now,
-as it can cause confusion later when dgit push produces its own tag.
-
 Now you can check out the branch tip,
 and do your substantive review.
 
@@ -245,16 +237,20 @@ or similar, to to the build, and then
 C<dgit -wgf [--quilt=...] push>
 to do the upload.
 
+Check whether the sponsee made a debian/I<version> tag.
+If they did,
+ensure you have their tag in the repository you are pushing from,
+or pass C<--no-dep14tag>.
+This avoids identically named, non-identical tags,
+which can be confusing.
+
 (It is possible to upload from
-the quilt-cache dgit view,
-but this will cause the debian/1.2.3-1 tag to be
-placed on this branch
-rather than the sponsee's working branch.
-Since this might be confusing,
-it is a good idea to switch back to the sponsee's view,
-after reviewing and before pushing.
-If you do want to upload from the quilt-cache dgit view,
-B<do not> pass the --quilt or --gbp or --dpm option again.)
+the quilt-cache dgit view.
+If you want to do this,
+B<do not> pass the C<--quilt> or C<--gbp> or C<--dpm> options again,
+and B<do> pass C<--no-dep14tag>,
+since the debian/I<version> tag
+should go on the sponsee's branch.)
 
 If this was the first upload done with dgit,
 you may need to pass
-- 
2.10.2

